sdk/java-sdk-spring/src/main/scala/kalix/spring/impl/KalixSpringApplication.scala

```diff
@@ -419,10 +431,24 @@ case class KalixSpringApplication(applicationContext: ApplicationContext, config
           }
           .foreach(messageCodec.registerTypeHints)
 
+        val field = lookupSuperclassField("kalixClient", workflow.getClass.getSuperclass)
+        field.setAccessible(true)
+        field.set(workflow, kalixClient(context))
+
         workflow
       })
   }
 
+  @tailrec
+  private def lookupSuperclassField(fieldName: String, clz: Class[_]): Field = {
+    try {
+      clz.getDeclaredField(fieldName)
+    } catch {
+      case e: NoSuchFieldException =>
```

Comment:
Why superclass here? Isn't `kalixClient` in the new `Action` or `Workflow` both of which are the direct parent from the user's component? I mean, we already call `getSuperclass` when calling to this method in the two cases above.

Reply:
We can only get the fields that are declared in a class. So, if I have `MyWorkflow extends Workflow`, to find the `kalixClient` field, we need to call `getDeclaredField` on the superclass of `MyWorkflow`, ie: `Workflow`.

So far so good. The first time we call this method we are already passing `workflow.getClass.getSuperclass`. 

But if we don't find it, we need to go one level down until we find it.

For example, image this case:
```java
abstract class MyBaseWorkflow extends Workflow {...}

class MyWorkflow extends MyBaseWorkflow {...}
```

The superclass of `MyWorkflow` is `MyBaseWorkflow` and therefore we won't find the `kalixClient` field on it. 
We need need to recover and dig deeper until we find the super-superclass `Workflow`. 

That's why this recursive lookup.

Reply:
Right. I see. I never thought the users would create an abstract `Action` or `Workflow` themselves. Seems a bit contrived but two more lines of codes to cover that case I think it's a sale.

Reply:
`try { clz.getDeclaredField(fieldName) } catch { case e: NoSuchFieldException => lookupSuperclassField(fieldName, clz.getSuperclass) }`

I can make it one line 😄